src/collective/volto/formsupport/restapi/services/submit_form/post.py

```diff
@@ -500,6 +519,21 @@ def attach_xml(self, msg):
 
     def store_data(self):
         store = getMultiAdapter((self.context, self.request), IFormDataStore)
-        res = store.add(data=self.filter_parameters())
+        data = {"form_data": self.filter_parameters()}
+
+        res = store.add(data=data)
         if not res:
             raise BadRequest("Unable to store data")
+
+        data.update(
+            {
+                "waiting_list": self.submit_limit is not None
+                and -1 < self.submit_limit < self.count_data()
+            }
+        )
+
+        return data
+
+    def count_data(self):
+        store = getMultiAdapter((self.context, self.request), IFormDataStore)
+        return store.count()
```

Comment:
quindi, aggiorni passando il valore della waiting list al momento della scrittura. ok

Reply:
Non so se sia il caso di mettere accrocchi che controllino l'esecuzione l'esecuzione di quel codice con dei lock... della serie, se due thead stanno inserendo l'ultima persona prima della waiting list? mi pare che non capiti nulla: ad ogni modo, uno dei due arriva prima dell'altro...

Reply:
dipende quando è critica la faccenda, per come la vedo io è comunque una domanda per qualcosa, quindi tutti i submit verranno poi revisionati da qualcuno in questo caso. Per questo motivo non la vedevo una cosa super critica